### DISCUSSION TITLE
Use Github to clone another branch from a remote repo into my clone of that repo

### DISCUSSION THREAD

**Discussion by richb-hanover** [community/community#61556] [New to GitHub] [answered]
I've been a contributor to a project XYZ for a while. I have cloned the `main` branch of the XYZ repo to my Github account, and make edits and PRs to contribute back. (I don't have write access to the XYZ repo.)

Someone created a new branch `new-branch` of the XYZ repo. How can I clone/pull/get that branch into my repo? I have not yet found a way to do this through the Github web GUI. Many thanks

**Comment by Beargruug** [comment]
you can change the branch on the left hand site of the github GUI. After selecting the branch you want you can clone the repo.

Here is a example: 

![image](https://github.com/community/community/assets/19775003/b081d348-23a0-42a1-b0cc-f2aab64aedd3)

**Comment by priyankarpal** [comment, marked as answer]
> I've been a contributor to a project XYZ for a while. I have cloned the `main` branch of the XYZ repo to my Github account, and make edits and PRs to contribute back. (I don't have write access to the XYZ repo.)
> 
> Someone created a new branch `new-branch` of the XYZ repo. How can I clone/pull/get that branch into my repo? I have not yet found a way to do this through the Github web GUI. Many thanks

- Delete the forked repo
- fork the repo again & make sure to uncheck `copy the main branch only`
 ![image](https://github.com/community/community/assets/88102392/e1f53bab-d1d2-4ade-b35e-355bc99fdbba)
- After that clone the forked repo to your local system
- Use `git branch --all` command to check all branches
- now switch branch using the command `git switch branch_name`


Hope it will help you

**Reply by richb-hanover** [reply]
Thanks. That is what I ultimately did - delete my forked repo (from Github), then clone again to my own Github account, with the "main branch only" box UNCHECKED.

I then cloned my repo to my laptop where I can work with gay abandon and only push the interesting changes to my repo on Github. I then can stay entirely within the Github ecosystem for official Push Requests. Thanks again
